### PR TITLE
WS-041: Admin locations endpoints

### PR DESCRIPTION
**Pull request by chiara-d697**
Create and Delete location endpoints have restricted access so only admins can delete or create a location. 
Update is not working but we have submitted this review as a draft so you can take a look.

### CODE REVIEW

**Comment by chiara-d697** [comment]
Update now working

File: backend/Controllers/LocationController.cs [hunk 1]
```diff
@@ -1,80 +1,281 @@
...
+
+            catch (InvalidOperationException)
+            {
+                return StatusCode(
+                    StatusCodes.Status401Unauthorized,
+                    "The given username is not valid"
+                );
+            }
+
+            if (user.Role == 0)
```
Comment (TimLeach635) [inline backend/Controllers/LocationController.cs:131]: Better to check that they're _not_ an admin - what if we add a new role that still shouldn't be able to add locations? A "guest", maybe? By default, they would be able to access this endpoint!

File: backend/Controllers/LocationController.cs [hunk 2]
```diff
@@ -1,80 +1,281 @@
...
+
+            if (!check)
+                return new UnauthorizedResult();
+
+            try
+            {
+                var location = _locations.CreateLocation(newLocation, user.Id);
+                return Created("/", newLocation);
+
+
```
Comment (TimLeach635) [inline backend/Controllers/LocationController.cs:149]: Whitespace!

File: backend/Controllers/LocationController.cs [hunk 3]
```diff
@@ -1,80 +1,281 @@
...
+            catch (InvalidOperationException)
+            {
+                return StatusCode(
+                    StatusCodes.Status401Unauthorized,
+                    "The given username is not valid"
+                );
+            }
+
+            if (user.Role == 0)
```
Comment (TimLeach635) [inline backend/Controllers/LocationController.cs:195]: Likewise here from above

File: backend/Controllers/LocationController.cs [hunk 4]
```diff
@@ -1,80 +1,281 @@
...
```
Comment (TimLeach635) [inline backend/Controllers/LocationController.cs:250]: ditto

File: backend/Models/Request/UpdateLocationRequest.cs [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+            Longitude = location.Longitude;
+            Name = location.Name;
+            Description = location.Description;
+            Sightings = location.Sightings;
+            Amenities = location.Amenities;
+        }
+        public UpdateLocationRequest()
+        {
+            
+        }
```
Comment (TimLeach635) [inline backend/Models/Request/UpdateLocationRequest.cs:34]: Don't change this, but you will never need this constructor - `UpdateLocationRequest` is parsed from HTTP requests sent to this API, so our code will only ever be converting _from_ this class, never to!

**Review by TimLeach635 — CHANGES_REQUESTED** [review]
Nice, just a few changes!

File: backend/Controllers/LocationController.cs (hunk 1 above)
Comment (DeepashriDali) [reply backend/Controllers/LocationController.cs]: Changed code to reflect not admin.

File: backend/Controllers/LocationController.cs (hunk 2 above)
Comment (DeepashriDali) [reply backend/Controllers/LocationController.cs]: Removed whitespace

File: backend/Models/Request/UpdateLocationRequest.cs (hunk 5 above)
Comment (DeepashriDali) [reply backend/Models/Request/UpdateLocationRequest.cs:34]: Left this in

**Review by TimLeach635 — APPROVED** [review]

**Pull request merged**